test/python/circuit/test_parameters.py

```diff
@@ -139,6 +140,41 @@ def test_get_parameters(self):
         self.assertIs(theta, next(iter(vparams)))
         self.assertEqual(rxg, vparams[theta][0][0])
 
+    def test_get_parameters_by_index(self):
+        """Test getting parameters by index"""
+        x = Parameter('x')
+        y = Parameter('y')
+        z = Parameter('z')
+        v = ParameterVector('v', 3)
+        qc = QuantumCircuit(1)
+        qc.rx(x, 0)
+        qc.rz(z, 0)
+        qc.ry(y, 0)
+        qc.u(*v, 0)
+        self.assertEqual(x, qc._parameter_table[0])
+        self.assertEqual(y, qc._parameter_table[2])
+        self.assertEqual(z, qc._parameter_table[1])
+        for i, vi in enumerate(v):
+            self.assertEqual(vi, qc._parameter_table[3+i])
+
+    def test_bind_parameters_anonymously(self):
+        """Test setting parameters by insertion order anonymously"""
+        x = Parameter('x')
+        y = Parameter('y')
+        z = Parameter('z')
+        v = ParameterVector('v', 3)
+        qc = QuantumCircuit(1)
+        qc.rx(x, 0)
+        qc.rz(z, 0)
+        qc.ry(y, 0)
+        qc.u(*v, 0)
+        params = [0.1 * i for i in range(len(qc.parameters))]
+        for assign_fun in ['bind_parameters', 'assign_parameters']:
+            with self.subTest(assign_fun=assign_fun):
+                bqc = getattr(qc, assign_fun)(params)
+                for value1, value2 in zip(bqc._parameter_table.values(), params):
+                    self.assertEqual(value1, value2)
+
```

Comment:
Can you also add a test covering appending and composing circuits/composite instructions?

Reply:
Let me know if the added tests cover this 🙂